go-controller/pkg/ovn/egressip.go

```diff
@@ -1516,23 +1535,44 @@ func (e *egressIPZoneController) deletePodEgressIPAssignment(egressIPName string
 			metrics.RecordEgressIPUnassign(duration)
 		}()
 	}
-
+	isLocalZoneEgressNode, loadedEgressNode := e.nodeZoneState.Load(status.Node)
 	isLocalZonePod, loadedPodNode := e.nodeZoneState.Load(pod.Spec.NodeName)
+	var nextHopIP string
+	// node may not exist - attempt to retrieve it
+	eNode, err := e.watchFactory.GetNode(status.Node)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("failed to add pod %s/%s because failed to lookup node %s: %v", pod.Namespace, pod.Name,
+			pod.Spec.NodeName, err)
+	}
+	if err == nil {
+		eIPIP := net.ParseIP(status.EgressIP)
+		isOVNManagedNetwork, err := util.IsOVNManagedNetwork(eNode, eIPIP)
+		if err != nil {
+			return fmt.Errorf("failed to delete egress IP %s (%s) because failed to determine if EgressIP is OVN managed: %v",
+				egressIPName, status.EgressIP, err)
+		}
+		nextHopIP, err = e.getNextHop(status.Node, status.EgressIP, egressIPName, isLocalZoneEgressNode, isOVNManagedNetwork)
+		if err != nil {
+			klog.Warningf("Unable to determine next hop for egress IP %s IP %s assigned to node %s: %v", egressIPName,
+				status.EgressIP, status.Node, err)
```

Comment:
are we missing a return after this? or an error? Otherwise nextHopIP will be "" and the function will proceed. Is that desired?

Reply:
nvm i see delete function called later checks if nexthopIP is empty

Reply:
if we fail to find the nextHop during the delete path, we just wipe all LRPs that have externalID with the name of the EIP obj below in `deleteReroutePolicyOps`